src/SourceControl-GitHub/APIWrapper.ts

```diff
@@ -0,0 +1,101 @@
+import { Octokit } from "octokit";
+import { Issue } from "./Issue";
+import { Milestone } from "./Milestone";
+import { Organization } from "./Organization";
+import { PullRequest } from "./PullRequest";
+import { Repository } from "./Repository";
+
+// authenticate via personal token for now
+const octokit = new Octokit({ auth: process.env._AUTHTOKEN });
+
+export class APIWrapper {
+  /**
+   * GetOrganization - queries GitHub for Organization information, resolving all
+   *                   sub info, such as Repositories, Issues, etc.
+   *
+   * @param  {type} orgName: string organization name
+   * @return {type}                 a GitHub organization
+   */
+  static async GetOrganization(orgName: string): Promise<Organization> {
+    const {
+      data: { id },
+    } = await octokit.rest.orgs.get({ org: orgName });
+
+    const repositories: { [key: string]: Repository } = {};
+
+    // if > 100 repos, will be paged
+    const respRepos = await octokit.rest.repos.listForOrg({ org: orgName });
+
+    for (let i = 0; i < respRepos.data.length; i++) {
+      let repoName = respRepos.data[i].name;
+      let repoId = respRepos.data[i].id.toString();
+      const repo = await this.GetRepository(orgName, repoName, repoId);
```

Comment:
It's been a while since I've done node/js, but how are they doing "concurrent" async calls these days?

Could be a potential perf improvement here to do something like & of course accounting for failures:

```js
await Promise.all(respRepos.data.map(r => this.GetRepository(orgName, repoName, repoId)))
```

Take this suggestion with a grain of salt it's been a while since I've been in js land... :D

Reply:
Echo the above ^^

Reply:
Right on!